packages/block-library/src/breadcrumbs/index.php

```diff
@@ -100,9 +100,25 @@ function render_block_core_breadcrumbs( $attributes, $content, $block ) {
 			$show_terms = $attributes['prefersTaxonomy'];
 		}
 
-		// Build breadcrumb trail.
+		// Add post type archive link if applicable.
+		$post_type_object = get_post_type_object( $post_type );
+		$archive_link     = get_post_type_archive_link( $post_type );
+		if ( $archive_link && untrailingslashit( home_url() ) !== untrailingslashit( $archive_link ) ) {
+			$label = $post_type_object->labels->archives;
+			if ( 'post' === $post_type ) {
+				$page_for_posts = get_option( 'page_for_posts' );
+				if ( $page_for_posts ) {
+					$label = block_core_breadcrumbs_get_post_title( $page_for_posts );
+				}
+			}
```

Comment:
How do you feel about this:

```suggestion
			if ( 'post' === $post_type && $page_for_posts = get_option( 'page_for_posts' ) ) {
				$label = block_core_breadcrumbs_get_post_title( $page_for_posts );
			}
```

Reply:
Actually I had to revert this, as linting doesn't allow it, even though is valid PHP.

Reply:
Ah, interesting. I used to love the brevity of the `if ( $var = something() )` construct, I assume linter's not happy because we're both assigning and checking at the same time (and it might look like an error - `=` vs `===`)

Reply:
> I used to love the brevity of the `if ( $var = something() )` construct

Same. I have this muscle memory of wrapping those assignments in an extra set of parentheses as a convention that it was an intentional assignment, and the vague memory that certain linters in certain projects accepted this (edit: ah, it's in C):

```php
if ( ( $foo = get_option( 'bar' ) ) ) {
    …
}
```

and it's frustrating that neither form works here.

I also caught up a few years ago with stuff happening in Pythonland, and they've had a specific assignment expression operator for a while now:

```python
if foo = get_option( 'bar' ):  # Syntax error (assignments are statements, not exprs)
    …

foo = get_option( 'bar' )      # Before
if foo:
    …

if foo := get_option( 'bar' ): # Nowadays, if you choose to
    …
```